src/cid.rs

```diff
@@ -56,11 +56,12 @@ impl Cid {
     }
 
     /// Create a new CID from a prefix and some data.
-    pub fn new_from_prefix(prefix: &Prefix, data: &[u8]) -> Cid {
-        let mut hash = prefix.mh_type.hasher().unwrap().digest(data);
-        if prefix.mh_len < hash.digest().len() {
-            hash = multihash::wrap(hash.algorithm(), &hash.digest()[..prefix.mh_len]);
-        }
+    pub fn new_from_prefix(prefix: &Prefix, digest: &[u8]) -> Cid {
```

Comment:
probably should update the doc comment to explain that the data must have been properly hashed.